src/lib/spinel/radio_spinel.hpp

```diff
@@ -353,6 +461,44 @@ template <typename InterfaceType> class RadioSpinel
     otError GetCoexMetrics(otRadioCoexMetrics &aCoexMetrics);
 #endif // OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
 
+    /**
+     * Get currently active interface.
+     *
+     * @param[out] aIid IID of the interface that owns the radio.
+     *
+     * @retval  OT_ERROR_NONE               Successfully got the property.
+     * @retval  OT_ERROR_RESPONSE_TIMEOUT   Failed due to no response received from the transceiver.
+     * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due to lack of the support in radio or platform supprts
+     *                                      all interfaces simultaneously
+     *                                      (i.e. no active/inactive interface concept in the platform level)
+     *
+     */
+    otError GetMultipanActiveInterface(spinel_iid_t *aIid);
+
+    /**
+     * Sets specified radio interface active
+     *
+     * This function allows selecting currently active radio interface on platforms that do not support parallel
+     * communication on multiple interfaces. I.e. if more than one interface is in receive state calling
+     * SetMultipanActiveInterface guarantees that specified interface will not be losing frames. This function
+     * returns if the request was received properly. After interface switching is complete SwitchoverDone callback is
+     * Invoked. Switching interfaces may take longer if aCompletePending is set true.
+     *
+     * @param[in] aIid              IID of the interface to set active.
+     * @param[in] aCompletePending  Set true if pending radio operation should complete first(Soft switch) or false if
+     * ongoing operations should be interrupted (Force switch).
+     *
+     * @retval  OT_ERROR_NONE               Successfully requested interface switch.
+     * @retval  OT_ERROR_BUSY               Failed due to another operation on going.
+     * @retval  OT_ERROR_RESPONSE_TIMEOUT   Failed due to no response received from the transceiver.
+     * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due to lack of support in radio for the given interface id or
+     *                                      platform supprts all interfaces simultaneously
+     *                                      (i.e. no active/inactive interface concept in the platform level)
+     * @retval  OT_ERROR_ALREADY            Given interface is already active.
```

Comment:
I think it would be more clear to end users of this API to return `OT_ERROR_ALREADY` on platforms that want to support discrete communications per radio instance instead of returning `OT_ERROR_NOT_IMPLEMENTED`.

Reply:
I will use OT_ERROR_INVALID_COMMAND for the same reason as in case of Get... function - so the caller clearly knows this call is not needed.